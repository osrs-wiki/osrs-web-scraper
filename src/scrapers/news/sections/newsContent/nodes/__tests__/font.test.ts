import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import fontParser from "../font";

describe("font node", () => {
  test("Font node should parse and render a 2-level header", () => {
    const root = parse(
      '<font style="font-family:Cinzel,serif;font-size:22px;font-weight:bold;color:#FFFFFF;text-shadow:1px 1px #121212;">test</font>'
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([fontParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Font node should parse and render a 3-level header", () => {
    const root = parse(
      '<font style="font-size:18px;font-weight:bold;color:#000000;">test</font>'
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([fontParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
