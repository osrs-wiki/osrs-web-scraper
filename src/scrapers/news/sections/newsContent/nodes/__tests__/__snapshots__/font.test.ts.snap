// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`font node Font node should parse and render a 2-level header 1`] = `"==test=="`;

exports[`font node Font node should parse and render a 3-level header 1`] = `"===test==="`;
